calico_cni_k8s_test.go

```diff
@@ -842,95 +847,229 @@ var _ = Describe("CalicoCni", func() {
 				})
 			})
 
-			Context("Using host-local IPAM: request an IP then release it, and then request it again", func() {
-				It("should successfully assign IP both times and successfully release it in the middle", func() {
-					netconfHostLocalIPAM := fmt.Sprintf(`
-				  {
-					"cniVersion": "%s",
-					"name": "net6",
-           			          "nodename_file_optional": true,
+			hostLocalIPAMConfigs := []struct {
+				description, config, unexpectedRoute string
+				expectedV4Routes, expectedV6Routes   []string
+			}{
+				{
+					description: "old-style inline subnet",
+					config: `
+					{
+					  "cniVersion": "%s",
+					  "name": "net6",
+					  "nodename_file_optional": true,
+					  "type": "calico",
+					  "etcd_endpoints": "http://%s:2379",
+					  "datastore_type": "%s",
+					  "ipam": {
+					    "type": "host-local",
+					    "subnet": "usePodCidr"
+					  },
+					  "kubernetes": {
+					   "k8s_api_root": "http://127.0.0.1:8080"
+					  },
+					  "policy": {"type": "k8s"},
+					  "log_level":"debug"
+					}`,
+					expectedV4Routes: []string{
+						regexp.QuoteMeta("default via 169.254.1.1 dev eth0"),
+						regexp.QuoteMeta("169.254.1.1 dev eth0 scope link"),
+					},
+					unexpectedRoute: regexp.QuoteMeta("10."),
+				},
+				{
+					// This scenario tests IPv4+IPv6 without specifying any routes.
+					description: "new-style with IPv4 and IPv6 ranges, no routes",
+					config: `
+					{
+					  "cniVersion": "%s",
+					  "name": "net6",
+					  "nodename_file_optional": true,
+					  "type": "calico",
+					  "etcd_endpoints": "http://%s:2379",
+					  "datastore_type": "%s",
+					  "ipam": {
+					    "type": "host-local",
+					    "ranges": [
+					       [
+					          {
+					            "subnet": "usePodCidr"
+					          }
+					       ],
+					       [
+					          {
+					            "subnet": "dead:beef::/96"
+					          }
+					       ]
+					    ]
+					  },
+					  "kubernetes": {
+					   "k8s_api_root": "http://127.0.0.1:8080"
+					  },
+					  "policy": {"type": "k8s"},
+					  "log_level":"debug"
+					}`,
+					expectedV4Routes: []string{
+						regexp.QuoteMeta("default via 169.254.1.1 dev eth0"),
+						regexp.QuoteMeta("169.254.1.1 dev eth0 scope link"),
+					},
+					expectedV6Routes: []string{
+						"dead:beef::. dev eth0  metric 256",
```

Comment:
Wondering what `.` will match here.  Is it just `0`?

Reply:
So is it OK to rely on that number being between 0 and 9?  You'd need e.g. `.*` to allow multiple digits?